enderio-base/src/main/java/com/enderio/base/common/item/darksteel/IDarkSteelItem.java

```diff
@@ -59,27 +63,49 @@ default void addCreativeItems(NonNullList<ItemStack> pItems, Item item) {
         pItems.add(is.copy());
     }
 
-    default void addUpgradeHoverTest(ItemStack pStack, List<Component> pTooltipComponents) {
-        if (DarkSteelUpgradeable.hasUpgrade(pStack, EmpoweredUpgrade.NAME)) {
-            String energy = EnergyUtil.getEnergyStored(pStack) + "/" + EnergyUtil.getMaxEnergyStored(pStack);
-            pTooltipComponents.add(new TextComponent(energy));
+    default void addCommonTooltips(ItemStack itemStack, @Nullable Player player, List<Component> tooltips) {
+    }
+
+    default void addBasicTooltips(ItemStack itemStack, @Nullable Player player, List<Component> tooltips) {
+        addCurrentUpgradeTooltips(itemStack, tooltips);
+    }
+
+    default void addDetailedTooltips(ItemStack itemStack, @Nullable Player player, List<Component> tooltips) {
+        addDurabilityTooltips(itemStack, tooltips);
+        addCurrentUpgradeTooltips(itemStack, tooltips);
+        addAvailableUpgradesTooltips(itemStack, tooltips);
+    }
+
+    default void addDurabilityTooltips(ItemStack itemStack,  List<Component> tooltips) {
+        if(itemStack.getDamageValue() > 0) {
+            String durability = (itemStack.getMaxDamage() - itemStack.getDamageValue()) + "/" + itemStack.getMaxDamage();
+            tooltips.add(new TextComponent(EnderCoreLang.DURABILITY.getString() + " " + durability).withStyle(ChatFormatting.GRAY));
+        }
+        if (DarkSteelUpgradeable.hasUpgrade(itemStack, EmpoweredUpgrade.NAME)) {
+            String energy = EnergyUtil.getEnergyStored(itemStack) + "/" + EnergyUtil.getMaxEnergyStored(itemStack) + " " + EnderCoreLang.ENERGY_ABBREVIATION.getString();
+            tooltips.add(new TextComponent(energy).withStyle(ChatFormatting.GRAY));
         }
+    }
 
-        var upgrades = DarkSteelUpgradeable.getUpgrades(pStack);
+    default  void addCurrentUpgradeTooltips(ItemStack itemStack, List<Component> tooltips) {
+        var upgrades = DarkSteelUpgradeable.getUpgrades(itemStack);
         upgrades
             .stream()
             .sorted(Comparator.comparing(INamedNBTSerializable::getSerializedName))
-            .forEach(upgrade -> pTooltipComponents.add(upgrade.getDisplayName()));
+            .forEach(upgrade -> tooltips.add(upgrade.getDisplayName().copy().withStyle(ChatFormatting.DARK_AQUA)));
+    }
 
-        var availUpgrades = DarkSteelUpgradeable.getUpgradesThatCanBeAppliedAtTheMoment(pStack);
+    default void addAvailableUpgradesTooltips(ItemStack itemStack, List<Component> tooltips) {
+        var availUpgrades = DarkSteelUpgradeable.getUpgradesThatCanBeAppliedAtTheMoment(itemStack);
+        EIOLang.DS_UPGRADE_AVAILABLE.copy().withStyle(ChatFormatting.YELLOW);
         if(!availUpgrades.isEmpty()) {
-            pTooltipComponents.add(TextComponent.EMPTY);
-            pTooltipComponents.add(EIOLang.DS_UPGRADE_AVAILABLE);
+            tooltips.add(TextComponent.EMPTY);
```

Comment:
Not sure we need this empty line, I'm not personally a fan. It wasn't there in 1.12 afaik and I did remove it in my #40 PR